log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/ValueResolver.java

```diff
@@ -20,24 +20,34 @@
  * #L%
  */
 
+import org.apache.logging.log4j.core.LogEvent;
+
 public interface ValueResolver {
 
     ValueResolver NO_OP = new ValueResolver() {
 
         @Override
-        public String resolve(String unresolved) {
+        public String resolve(String unresolved, LogEvent logEvent) {
             return unresolved;
         }
 
         @Override
-        public String resolve(VirtualProperty property) {
+        public String resolve(VirtualProperty property, LogEvent logEvent) {
             return property.getValue();
         }
 
     };
 
-    String resolve(String unresolved);
+    default String resolve(String unresolved) {
+        return resolve(unresolved, null);
+    }
+
+    String resolve(String unresolved, LogEvent logEvent);
```

Comment:
Unsure about this: I added the log event as an overloaded method because a) the resolve method is called in a lot of places including in many unit tests, so this way there are fewer changes b) it's more obvious that the log event is optional this way.